applications/nrf5340_audio/src/bluetooth/bt_stream/broadcast/broadcast_source.h

```diff
@@ -77,22 +77,79 @@
 #error Unsupported LC3 codec preset for broadcast
 #endif /* CONFIG_BT_AUDIO_BROADCAST_CONFIGURABLE */
 
+#define BROADCAST_SOURCE_ADV_NAME_MAX	  (32)
+#define BROADCAST_ADV_ID_START		  (BT_UUID_SIZE_16)
+#define BROADCAST_SOURCE_PER_ADV_BUF_SIZE (128)
+
+#if (CONFIG_AURACAST)
+/* Size of the Public Broadcast Announcement set buffer in bytes.
+ *
+ * @note: Make sure the PBA buffer is large enough for a 16bit UUID and meta data
+ *        (any addition to pba_buf requires an increase of this value).
+ */
+#define BROADCAST_SOURCE_PBA_SIZE (BT_UUID_SIZE_16 + 8)
+#endif /* (CONFIG_AURACAST) */
+
 /**
- * @brief	Get the data to advertise.
+ * @brief	Advertising data for broadcast source.
+
+ */
+struct broadcast_source_ext_adv_data {
+	/* Broadcast Audio Streaming UUIDs. */
+	struct net_buf_simple *uuid_buf;
+
+	/* Broadcast Audio Streaming Endpoint advertising data. */
+	uint8_t brdcst_id_buf[BT_UUID_SIZE_16 + BT_AUDIO_BROADCAST_ID_SIZE];
+
+	/* Buffer for Appearance. */
+	uint8_t brdcst_appearance_buf[BT_UUID_SIZE_16];
+
+	/* Broadcast name, must be between 4 and 32 UTF-8 encoded characters in length. */
+	uint8_t brdcst_name_buf[BROADCAST_SOURCE_ADV_NAME_MAX];
+
+#if (CONFIG_AURACAST)
+	/* Public Broadcast Announcement set buffer. */
+	uint8_t pba_buf[BROADCAST_SOURCE_PBA_SIZE];
+#endif /* (CONFIG_AURACAST) */
+};
+
+/**
+ * @brief	Periodic advertising data for broadcast source.
+ */
+struct broadcast_source_per_adv_data {
+	/* Buffer for periodic advertising data */
+	struct net_buf_simple *base_buf;
+};
```

Comment:
It is not clear to me if this will be the only element and it makes the API's for consistent, but yes if this is the only element then it could be removed.
Having a structure here allows us to add more data (e.g. length) without effecting the API and thus the customers code.